### DISCUSSION TITLE
Bazel build

### DISCUSSION THREAD

**Discussion by malkia** [ArthurSonzogni/FTXUI#800] [General]
Just for fun, I've forked the repo and made a bazel build. Mainly tested on Windows with both MSVC and clang-cl

https://github.com/malkia/FTXUI

**Comment by ArthurSonzogni** [comment]
Thanks @malkia !

This is looking good to me!
Would you like to submit a PR to integrate this?

Some quick suggestions:
- I suspect some changes will be reverted by IWUY: (`./tools/iwyu.sh`). We should find a way to configure IWYU and Bazel so that they won't contradict each other.
- Bazel lock files. I am not a huge fan of committing generated files. Can we avoid adding it to the repository?

**Comment by malkia** [comment]
Hah! Awesome that you are willing to accept that. I need a bit more time to strengthen this up, make up really good PR and come back at you - also look int IWYU integration and few other things! Happy New Year!!!

**Comment by Clumsyndicate** [comment]
I'm using a bazel setup too. Do you know how to integrate ftxui into a bazel project with all the source code?
Currently can only see the headers and the object files in my bazel repo. Would be much more convenient to be able to build from source.

Thanks!

**Reply by malkia** [reply]
I actually did it here - https://github.com/malkia/opentelemetry-cpp/blob/main/x/BUILD.bazel#L70 - Sorry, it's `Windows` only - but nothing stops it from working on other systems (it might actually works). I wanted some terminal UI for a testing I've been doing with the OTEL project - and having used FTXUI for internal projects before I've decided this is the way to go (though still bit rusty).

The use is here - https://github.com/malkia/opentelemetry-cpp/blob/main/WORKSPACE#L84 - but you can do it other ways too. Ideally it should use the newer `bzlmod` - and eventually it'll come. There might be much simpler integrations than mine also.

**Comment by ArthurSonzogni** [comment]
@malkia are you still planning to merge the Bazel config?
